rumydata/table.py

```diff
@@ -283,12 +283,12 @@ def _check(self, filepath: Union[str, Path], **kwargs) -> Union[ex.FileError, No
 
                 if re:
                     e.append(re)
-                    if rix == 0:  # if header error present, stop checking rows
+                    if rix == (0 + self.skip_rows) and self.layout.no_header is False:  # if header error present, stop checking rows
                         break
                     if len(e) > self.max_errors:
                         e.append(max_error_rule._exception_msg())
                         break
-                if rix > 0:
+                if rix > 0 or self.layout.no_header is True:
```

Comment:
In looking at this, I think there's a risk that we would skip rows (as intended), but then cache their values as part of a the column cache that is used for column rules. I'll try to write some tests that will look for this.

Reply:
Nevermind, this totally works, and the existing tests cover it. I was worried that the combined logic of skipping rows and skipping headers might lead to unexpected results in the column cache, but I was wrong.

Reply:
Wait, wait, no I found it. So there is a problem with this logic. You can demonstrate it like so:
```python
from pathlib import Path
from tempfile import TemporaryDirectory

from rumydata.field import Text
from rumydata.rules.column import Unique
from rumydata.table import Layout, CsvFile

with TemporaryDirectory() as tmpdir:
    p = Path(tmpdir, 'my_file.csv')
    p.write_text('\n'.join(['a', 'a', 'a', 'a']))
    layout = Layout({'a': Text(2, rules=[Unique()])})
    csv = CsvFile(layout, skip_rows=2)
    csv.check(p)
```
In the code above we write a CSV file with the value 'a' repeated on four rows. We define the layout as a *column* named `a`. Then we tell the CsvFile to skip the first two rows. The result is that we should interpret the data as having a column named `a`, with a single row with the value `a`. This should not trigger a violation of the `Unique` rule, but it does, because the column name gets cached as part of the rest of the columnar values, and then evaluated against the unique check.